In [ ]:
!pip install python-dotenv
!pip install python-decouple
# !pip install --upgrade genai


In [ ]:
import os
import numpy as np
from sklearn.metrics.pairwise import cosine_similarity
from decouple import config
from dotenv import load_dotenv


In [ ]:
from decouple import config
import google.generativeai as genai


def generate_embeddings(text: str) -> list[float] | None:
    api_key = config("GEMINI_API_KEY")

    genai.configure(api_key=api_key)

    result = genai.embed_content(
        model=config("EMBEDDINGS_MODEL"),
        content=text,
    )
    embeddings = result.get("embedding")

    return embeddings

In [ ]:
def similaridade_de_cosseno(texto1: str, texto2: str) -> float:

    embeddings1 = generate_embeddings(texto1)
    embeddings2 = generate_embeddings(texto2)

    if embeddings1 is None or embeddings2 is None:
        raise ValueError("Erro ao gerar embeddings para os textos.")

    similaridade = cosine_similarity([embeddings1], [embeddings2])

    return similaridade[0][0]

In [ ]:
# Exemplo de uso da função
texto1 = "Yes, passive acoustic mapping (PAM) is a promising tool for monitoring acoustic cavitation activities in ultrasound therapy.  It allows continuous monitoring without interrupting therapeutic pulses and has shown efficacy in various ultrasound treatments.  While time exposure acoustics (TEA) is efficient, it has limitations in spatial resolution and artifact suppression. Data-adaptive beamformers improve image quality but are computationally expensive.  A deep beamformer approach offers a significant reduction in computational cost while maintaining image quality comparable to data-adaptive methods."
texto2 = "Passive acoustic mapping (PAM) is a promising tool for monitoring acoustic cavitation activities in ultrasound therapy applications."

similaridade = similaridade_de_cosseno(texto1, texto2)
print(f"Similaridade de cosseno: {similaridade:.4f}")


Similaridade de Consenso: 0.8207


```
1: Os vetores são idênticos (ou seja, a direção é exatamente a mesma).
0: Não há similaridade (os vetores são ortogonais, ou seja, não têm correlação).
-1: Os vetores são opostos (ou seja, estão em direções opostas).
```



What important role does private inference (PI) play?
```
# Resposta trabalho

Based on the provided text, private inference (PI) plays an important role in protecting the privacy of agent evidence in collective decision-making.  The direct sharing of raw evidence leads to participants' preference leakage, so PI methods are necessary to prevent this.  The paper describes a privacy-preserving distributed credible evidence fusion algorithm (PCEF) that uses PI to guarantee evidence privacy and prevent raw evidence from being inferred.

# Pergunta tirada do artigo TRUNCFORMER: PRIVATE LLM INFERENCE USING ONLY TRUNCATIONS`

Private inference (PI) plays an important role by enabling machine learning models to make inferences on sensitive or private data without revealing this information. This is crucial for ensuring data privacy and security in applications where the protection of personal information is essential, such as in financial services, healthcare, and other sectors dealing with confidential data.
```

Similaridade de Cosseno: 0.7635


What is the promising strategy for teaching robots a wide range of complex skills that recent works have demonstrated?
```
# Resposta trabalho

Recent work has demonstrated that a promising strategy for teaching robots a wide range of complex skills is to use autoregressive generative pre-training on latent motion token sequences to learn motion priors from videos, followed by co-fine-tuning on action-labeled data for robot control.

# Pergunta tirada do artigo Environment Curriculum Generation
via Large Language Models

Recent works have demonstrated that a promising strategy for teaching robots a wide range of complex skills is to train them on a curriculum of progressively more challenging environments.
```

Similaridade de Cosseno: 0.7110

Randomized rounding is a technique originally used for what?
```
# Resposta trabalho

The provided text states that randomized rounding was originally a technique used for  achieving performative stability within the Repeated Risk Minimization (RRM) framework.

# Pergunta tirada do artigo Randomized Rounding Approaches to Online
Allocation, Sequencing, and Matching

Randomized rounding is a technique originally used to approximate hard offline discrete optimization problems from a mathematical programming relaxation.
```

Similaridade de Cosseno: 0.7517

What is a fundamental component of transformers?
```
# Resposta trabalho

Based on the provided text, the main components of transformers are the encoder and the decoder.  These are stacked multiple times.  The encoder embeds a sentence into a set of numbers, while the decoder generates text based on the embedding of an input sentence.  Another important component is the multi-head attention mechanism, which computes similarity scores between words in a sentence.

# Pergunta tirada do artigo THE ASYMPTOTIC BEHAVIOR OF ATTENTION IN TRANSFORMERS

A key component of transformers is the attention mechanism orchestrating how each token influences the propagation of every other token through a transformer.
```

Similaridade de Cosseno: 0.7983

What is Truck Attention-QMIX (TA-QMIX)?
```
# Resposta trabalho

Truck Attention-QMIX (TA-QMIX) is a Multi-Agent Deep Reinforcement Learning framework for efficient online decision policy training in truck platooning coordination.  It uses an attention mechanism to improve the representation of truck fuel gains and delay times, providing explicit cooperation information during training to encourage cooperation between trucks.  The framework uses centralized training and distributed execution, allowing trucks to make online decisions using only local information.  In experiments with 5,000 trucks, it achieved an average fuel savings of 19.17% with an average delay of only 9.57 minutes per truck and a decision time of 0.001 seconds.

# Pergunta tirada do artigo Multi-Agent Deep Reinforcement Learning for
Distributed and Autonomous Platoon Coordination via Speed-regulation over Large-scale Transportation Networks

Truck Attention-QMIX (TA-QMIX) is a deep reinforcement learning multi-agent framework developed to solve the problem of truck platoon coordination in large-scale transportation networks. The main objective is to optimize overall transportation efficiency by promoting cooperation among trucks and balancing fuel savings with delays in departure times and speed.
```

Similaridade de Cosseno: 0.9537

Is passive acoustic mapping (PAM) a promising tool
```
# Resposta trabalho

Yes, passive acoustic mapping (PAM) is a promising tool for monitoring acoustic cavitation activities in ultrasound therapy.  It allows continuous monitoring without interrupting therapeutic pulses and has shown efficacy in various ultrasound treatments.  While time exposure acoustics (TEA) is efficient, it has limitations in spatial resolution and artifact suppression. Data-adaptive beamformers improve image quality but are computationally expensive.  A deep beamformer approach offers a significant reduction in computational cost while maintaining image quality comparable to data-adaptive methods.

# Pergunta tirada do artigo Switchable deep beamformer for high-quality and real-time passive acoustic mapping

Passive acoustic mapping (PAM) is a promising tool for monitoring acoustic cavitation activities in ultrasound therapy applications.
```

Similaridade de Cosseno: 0.8207